internal/export/dashboards.go

```diff
@@ -0,0 +1,117 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package export
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/common"
+	"github.com/elastic/elastic-package/internal/kibana"
+	"github.com/elastic/elastic-package/internal/logger"
+	"github.com/elastic/elastic-package/internal/packages"
+)
+
+// Dashboards method exports selected dashboards with references objects. All Kibana objects are saved to local files
+// in appropriate directories.
+func Dashboards(kibanaClient *kibana.Client, dashboardsIDs []string) error {
+	packageRoot, err := packages.MustFindPackageRoot()
+	if err != nil {
+		return errors.Wrap(err, "locating package root failed")
+	}
+	logger.Debugf("Package root found: %s", packageRoot)
+
+	m, err := packages.ReadPackageManifest(filepath.Join(packageRoot, packages.PackageManifestFile))
+	if err != nil {
+		return errors.Wrapf(err, "reading package manifest failed (path: %s)", packageRoot)
+	}
+
+	objects, err := kibanaClient.Export(dashboardsIDs)
+	if err != nil {
+		return errors.Wrap(err, "exporting dashboards using Kibana client failed")
+	}
+
+	ctx := &transformationContext{
+		packageName: m.Name,
+	}
+
+	objects, err = applyTransformations(ctx, objects)
+	if err != nil {
+		return errors.Wrap(err, "can't transform Kibana objects")
+	}
+
+	err = saveObjectsToFiles(packageRoot, objects)
+	if err != nil {
+		return errors.Wrap(err, "can't save Kibana objects")
+	}
+	return nil
+}
+
+func applyTransformations(ctx *transformationContext, objects []common.MapStr) ([]common.MapStr, error) {
+	return transformObjects(ctx, objects,
+		filterUnsupportedTypes,
+		decodeObject,
+		stripObjectProperties,
+		standardizeObjectProperties,
+		standardizeObjectID)
+}
```

Comment:
Nit: this is probably just personal preference so feel free to ignore this comment. 

Looking at the signature and intended usage for `transformObjects` it seems like a good candidate for a fluent pattern. So something like:

```
return objectTransformer().
  withContext(ctx).
  withTransforms(filterUnsupportedTypes, decodeObject, ...).
  transform(objects)
```

Reply:
Good idea. I modified the source code to the suggested form.